From 6cf67b41287e1c6e5b1e7d968ffac912b96c1d19 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Tue, 8 Mar 2022 13:53:09 -0700
Subject: [PATCH 09/17] crypto: virtio - Select new dependencies

With ARCH=riscv defconfig, there are errors at link time:

  virtio_crypto_akcipher_algs.c:(.text+0x3ea): undefined reference to `mpi_free'
  virtio_crypto_akcipher_algs.c:(.text+0x48a): undefined reference to `rsa_parse_priv_key'
  virtio_crypto_akcipher_algs.c:(.text+0x4bc): undefined reference to `rsa_parse_pub_key'
  virtio_crypto_akcipher_algs.c:(.text+0x4d0): undefined reference to `mpi_read_raw_data'
  virtio_crypto_akcipher_algs.c:(.text+0x960): undefined reference to `crypto_register_akcipher'
  virtio_crypto_akcipher_algs.c:(.text+0xa3a): undefined reference to `crypto_unregister_akcipher'

The virtio crypto driver started making use of certain libraries and
algorithms without selecting them. Do so to fix these errors.

Fixes: 8a75f36b5d7a ("virtio-crypto: implement RSA algorithm")
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/r/20220308205309.2192502-1-nathan@kernel.org
---
 drivers/crypto/virtio/Kconfig | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/crypto/virtio/Kconfig b/drivers/crypto/virtio/Kconfig
index b894e3a8be4f..5f8915f4a9ff 100644
--- a/drivers/crypto/virtio/Kconfig
+++ b/drivers/crypto/virtio/Kconfig
@@ -3,8 +3,11 @@ config CRYPTO_DEV_VIRTIO
 	tristate "VirtIO crypto driver"
 	depends on VIRTIO
 	select CRYPTO_AEAD
+	select CRYPTO_AKCIPHER2
 	select CRYPTO_SKCIPHER
 	select CRYPTO_ENGINE
+	select CRYPTO_RSA
+	select MPILIB
 	help
 	  This driver provides support for virtio crypto device. If you
 	  choose 'M' here, this module will be called virtio_crypto.
-- 
2.35.1

